protocol/river-control-unstable-v1.xml

```diff
@@ -37,49 +29,421 @@
+      <arg name="alpha" type="fixed"/>
+    </request>
+
+    <request name="border_color_focused">
+      <description summary="set the border color of focused views"/>
+      <arg name="red" type="fixed"/>
+      <arg name="green" type="fixed"/>
+      <arg name="blue" type="fixed"/>
+      <arg name="alpha" type="fixed"/>
+    </request>
+
+    <request name="border_color_unfocused">
+      <description summary="set the border color of unfocused views"/>
+      <arg name="red" type="fixed"/>
+      <arg name="green" type="fixed"/>
+      <arg name="blue" type="fixed"/>
+      <arg name="alpha" type="fixed"/>
+    </request>
+
+    <request name="border_width">
+      <description summary="set the border width"/>
+      <arg name="pixels" type="uint" summary="logical pixels"/>
+    </request>
+
+    <request name="outer_padding">
+      <description summary="set the padding around the edge of the screen"/>
+      <arg name="pixels" type="uint" summary="logical pixels"/>
+    </request>
+
+    <request name="view_padding">
+      <description summary="set the padding between views"/>
+      <arg name="pixels" type="uint" summary="logical pixels"/>
+    </request>
+
+    <request name="declare_mode">
+      <description summary="declare a new mode">
+        Modes are used when defining mappings. If a mode with the given name
+        already exists this request does nothing.
+      </description>
+      <arg name="name" type="string" summary="name of the new mode"/>
+    </request>
+
+    <request name="enter_mode">
+      <description summary="enter the given mode"/>
+      <arg name="seat" type="object" interface="wl_seat"/>
+      <arg name="name" type="string" summary="name of the new mode"/>
+      <arg name="callback" type="new_id" interface="zriver_enter_mode_callback_v2"/>
+    </request>
+
+    <enum name="modifiers" bitfield="true">
+      <entry name="none" value="0"/>
+      <entry name="shift" value="1"/>
+      <entry name="caps" value="2"/>
+      <entry name="ctrl" value="4"/>
+      <entry name="alt" value="8"/>
+      <entry name="mod2" value="16"/>
+      <entry name="mod3" value="32"/>
+      <entry name="logo" value="64"/>
+      <entry name="mod5" value="128"/>
+    </enum>
+
+    <request name="map">
+      <description summary="create, remove, or modify a mapping">
+        Create a mapping, overwriting an existing mapping if one already
+        exists for the mode/modifiers/key/state combination. If command is
+        null, instead remove the mapping if it exists.
+
+        A complete list of XKB key names for the key argument can usually be
+        found at /usr/include/xkbcommon/xkbcommon-keysyms.h.
+      </description>
+      <arg name="mode" type="string"/>
+      <arg name="modifiers" type="uint" enum="modifiers"/>
+      <arg name="key" type="string" summary="XKB key name"/>
+      <arg name="state" type="uint" enum="wl_keyboard.key_state"/>
+      <arg name="command" type="string" allow-null="true"/>
```

Comment:
Just from reading the protocol it is not clear what `command` is supposed to be. Most people would probably interpret it as a shell command. Also it should be documented somewhere what strings make river do what.

Reply:
It is just a shell command, though I realize I should probably have documented that better. In order to remove all parsing code from the compositor, my current plan is to have riverctl pass e.g. `riverctl focus-view next` as the command argument to map when invoked with `riverctl map normal Mod4 J focus-next`.